TODO: normalize path case in Windows

Error found in: https://github.com/JuliaDiffEq/diffeqpy/pull/37

```
E   RuntimeError: It seems your Julia and PyJulia setup are not supported.
E   
E   Julia interpreter:
E       julia
E   Python interpreter and libpython used by PyCall.jl:
E       C:\projects\diffeqpy\.tox\py\scripts\python.exe
E       C:\projects\diffeqpy\.tox\py\scripts\python36.dll
E   Python interpreter used to import PyJulia and its libpython.
E       c:\projects\diffeqpy\.tox\py\scripts\python.exe
E       c:\projects\diffeqpy\.tox\py\scripts\python36.dll
E   
E   In Julia >= 0.7, above two paths to `libpython` have to match exactly
E   in order for PyJulia to work.
```

--- https://ci.appveyor.com/project/ChrisRackauckas/diffeqpy/builds/21168243/job/684yeqmybv84d6o5

Notice uppercase `C:` and lowercase `c:`.

We need to use https://docs.python.org/3/library/os.path.html#os.path.normcase

--- a/julia/core.py
+++ b/julia/core.py
@@ -42,7 +42,7 @@ except ImportError:
 # this is python 3.3 specific
 from types import ModuleType
 
-from .find_libpython import find_libpython, linked_libpython, normalize_path
+from .find_libpython import find_libpython, linked_libpython
 from .options import JuliaOptions, parse_jl_options
 
 #-----------------------------------------------------------------------------
@@ -434,7 +434,7 @@ class JuliaInfo(object):
         return is_compatible_exe(self.libpython_path)
 
 
-def is_compatible_exe(libpython):
+def is_compatible_exe(jl_libpython):
     """
     Determine if `libpython` is compatible with this Python.
 
@@ -443,13 +443,11 @@ def is_compatible_exe(libpython):
     this function returns `True`, PyJulia use the same precompilation cache
     of PyCall.jl used by Julia itself.
     """
-    logger.debug("libpython = %s", libpython)
     py_libpython = linked_libpython()
-    jl_libpython = normalize_path(libpython)
     logger.debug("py_libpython = %s", py_libpython)
     logger.debug("jl_libpython = %s", jl_libpython)
     dynamically_linked = py_libpython is not None
-    return dynamically_linked and py_libpython == jl_libpython
+    return dynamically_linked and os.path.samefile(py_libpython, jl_libpython)
     # `py_libpython is not None` here for checking if this Python
     # executable is dynamically linked or not (`py_libpython is None`
     # if it's statically linked).  `jl_libpython` may be `None` if
@@ -774,7 +772,7 @@ def raise_separate_cache_error(
         runtime=runtime,
         jlinfo=jlinfo,
         py_libpython=find_libpython(),
-        jl_libpython=normalize_path(jlinfo.libpython_path),
+        jl_libpython=jlinfo.libpython_path,
         sys=sys)
     raise RuntimeError(message)
 

--- a/julia/core.py
+++ b/julia/core.py
@@ -39,6 +39,16 @@ except ImportError:
     # shutil.which when used with single argument.
     from distutils.spawn import find_executable as which
 
+try:
+    from os.path import samefile
+except ImportError:
+    # For Python < 3.2 in Windows:
+    def samefile(f1, f2):
+        a = os.path.realpath(os.path.normcase(f1))
+        b = os.path.realpath(os.path.normcase(f2))
+        return a == b
+
+
 # this is python 3.3 specific
 from types import ModuleType
 
@@ -447,7 +457,7 @@ def is_compatible_exe(jl_libpython):
     logger.debug("py_libpython = %s", py_libpython)
     logger.debug("jl_libpython = %s", jl_libpython)
     dynamically_linked = py_libpython is not None
-    return dynamically_linked and os.path.samefile(py_libpython, jl_libpython)
+    return dynamically_linked and samefile(py_libpython, jl_libpython)
     # `py_libpython is not None` here for checking if this Python
     # executable is dynamically linked or not (`py_libpython is None`
     # if it's statically linked).  `jl_libpython` may be `None` if

--- a/julia/core.py
+++ b/julia/core.py
@@ -434,12 +434,6 @@ class JuliaInfo(object):
         return is_compatible_exe(self.libpython_path)
 
 
-def is_same_path(a, b):
-    a = os.path.realpath(os.path.normcase(a))
-    b = os.path.realpath(os.path.normcase(b))
-    return a == b
-
-
 def is_compatible_exe(libpython):
     """
     Determine if `libpython` is compatible with this Python.

--- a/julia/find_libpython.py
+++ b/julia/find_libpython.py
@@ -88,6 +88,8 @@ def _linked_libpython_unix():
     if retcode == 0:  # means error
         return None
     path = os.path.realpath(dlinfo.dli_fname.decode())
+    if not os.path.exists(path):
+        return None
     if path == os.path.realpath(sys.executable):
         return None
     return path

